lib/xeroizer/generic_application.rb

```diff
@@ -72,7 +72,7 @@ def initialize(consumer_key, consumer_secret, options = {})
         @before_request = options.delete(:before_request)
         @after_request = options.delete(:after_request)
         @around_request = options.delete(:around_request)
-        @client = OAuth.new(consumer_key, consumer_secret, options.merge({default_headers: default_headers}))
+        @client = client
```

Comment:
Consider checking client.kind_if?(OAuth) or similar; raising an argument error if given strings accidentally.

Reply:
We can do that.

Reply:
Resolved above